### PR TITLE
tests: add xtimer_hang

### PR DESCRIPTION
**Pull request by kaspar030**
Test case for #4060, fixed on samr21-xpro by #4061.

### CODE REVIEW

**Comment by daniel-k** [comment]
Works for me on `samr21-xpro` (i.e. ahngs) and I can confirm that #4061 fixes it. ACK.

**Comment by kaspar030** [comment]
- added percentage output, squashed, triggered CI

**Comment by kaspar030** [comment]
Triple green -> go.

File: tests/xtimer_hang/main.c [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ * @}
+ */
+#include <stdio.h>
+
+#include "xtimer.h"
+#include "thread.h"
+
+#define TEST_SECONDS        (10U)
+#define TEST_TIME           (TEST_SECONDS * SEC_IN_USEC)
+#define STACKSIZE_TIMER     (1024)
```
Comment (Kijewski) [inline tests/xtimer_hang/main.c]: Hm.

File: tests/xtimer_hang/main.c [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+char stack_timer1[STACKSIZE_TIMER];
+char stack_timer2[STACKSIZE_TIMER];
+
+long unsigned int count = 0;
+
+void* timer_func1(void* arg)
+{
+    (void)arg;
+    while(1)
```
Comment (Kijewski) [inline tests/xtimer_hang/main.c]: Hm

**Pull request merged**